Enterprise-CMCS/managed-care-review
# MCR-3988 Update the Dates in Documents Table to use Contract data

## Summary
This was was needed so that Date Added and the "NEW" tag works in the `UploadedDocumentsTable` on V2 pages we have built with the Contract and Rate API. 

Changes:
- Add `dateAdded` field to backend on each `GenericDocument`. This required some type weaving and test fixing because we have a lot of strong assumptions about the form data and how it works in the application.
- Add `prismaContractRateAdaptors` to share the prisma operations used on .create and .update between database functions
- Create a new shared zod types for `ContractEditableType` and `RateEditableType` before these types were defined in random places in postgres and graphql code. 
- Remove `isSubmitting` prop on `<UploadedDocumentsTable />`, just rely on `isEditing`. Also make this component take in the `previousSubmissionDate` as a prop and accept documents that already know their own `dateAdded`.

#### Related issues
https://jiraent.cms.gov/browse/MCR-3988

#### Screenshots

#### Test cases covered
- Existing tests still pass including uploaded documents table

## QA guidance
- Submit a package
- Log in as a CMS user and check the submission summary page documents tables. Those tables should have a Date Added column with a date in place. 
- Date added is present  BOTH on the Contract Details summary and on the Rate Details summary  (one is a v2 page one is not)
